packages/core/src/components/forms/numericInputUtils.ts

```diff
@@ -30,7 +30,7 @@ function getDecimalSeparator(locale: string) {
 }
 
 export function toLocaleString(num: number, locale: string = "en-US") {
-    return sanitizeNumericInput(num.toLocaleString(locale), locale);
+    return sanitizeNumericInput(num.toLocaleString(locale, { roundingPriority: "morePrecision" } as any), locale);
```

Comment:
@Rafael-Martins why do you need to cast `as any`? I'd prefer to avoid that cast if we can. If it's _absolutely necessary_, then it should be accompanied with a `// HACKHACK: explanation` comment on the previous line.